src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1852,13 +1837,85 @@
     </Field>
     <Field Name="Value" Type="ExpressionSyntax"/>
   </Node>
+
+  <AbstractNode Name="VariableComponentSyntax" Base="CSharpSyntaxNode">
+  </AbstractNode>
+  <Node Name="TypedVariableComponentSyntax" Base="VariableComponentSyntax">
+    <Kind Name="TypedVariableComponent"/>
+    <Field Name="Type" Type="TypeSyntax"/>
+    <Field Name="Designation" Type="VariableDesignationSyntax"/>
+  </Node>
+  <Node Name="TupleDeconstructionVariableComponentSyntax" Base="VariableComponentSyntax">
+    <Kind Name="TupleDeconstructionVariableComponent"/>
+    <Field Name="OpenParenToken" Type="SyntaxToken">
+      <Kind Name="OpenParenToken"/>
+    </Field>
+    <Field Name="Variables" Type="SeparatedSyntaxList&lt;VariableComponentSyntax&gt;"/>
+    <Field Name="CloseParenToken" Type="SyntaxToken">
+      <Kind Name="CloseParenToken"/>
+    </Field>
+  </Node>
+  <!--
+  <Node Name="WildcardVariableComponentSyntax" Base="VariableComponentSyntax">
+    <Kind Name="WildcardVariableComponent"/>
+    <Field Name="AsteriskToken" Type="SyntaxToken">
+      <Kind Name="AsteriskToken"/>
+    </Field>
+  </Node>
+  -->
+
+  <AbstractNode Name="VariableDesignationSyntax" Base="CSharpSyntaxNode">
+  </AbstractNode>
+  <Node Name="SingleVariableDesignationSyntax" Base="VariableDesignationSyntax">
```

Comment:
can we just make VariableDeclaratorSyntax a VariableDesignationSyntax?  Then we don't need SingleVariableDesignation.  And we'll be able to handle "type v = expr".  It also provides a nice a bridge between the existing world and the new world.